budgettool/transaction.py

```diff
@@ -0,0 +1,45 @@
+"""Containers for metadata associated with a transaction
+or entry in the ledger.
+"""
+from . import schedulers
+
+class Transaction:
+    """Minimum data associated with all transactions.
+    """
+    def __init__(self, name, category, amount):
+        self.name = name
+        self.category = category
+        self.amount = amount
+
+
+class TemplateTransaction:
+    """Template from which transactions can be generated on a schedule.
+    """
+    @staticmethod
+    def from_dict(data):
+        """Generate a transaction template from the standard dictionary storage format.
+        """
+        name     = data['name']
+        category = data['category']
+        amount   = data['amount']
+        schedule = schedulers.from_dict(data['schedule'])
+        return TemplateTransaction(
+            name=name, category=category, amount=amount, schedule=schedule)
+
+    def __init__(self, name, category, amount, schedule):
+        self.transaction = Transaction(name, category, amount)
+        self.schedule = schedule
+
+    def __iter__(self):
```

Comment:
Once we fix issue #4 , I would suggest using __iter__ to generate all the transactions. That would make the forecast function a tiny bit simpler.